include/seastar/core/simple-stream.hh

```diff
@@ -396,15 +405,18 @@ private:
         }
     }
     fragmented_memory_input_stream(Iterator it, simple bv, size_t size)
-        : _it(it), _current(bv), _size(size) { }
+        : _it(it), _current(bv), _size(size), _initial_size(size) { }
     friend class fragmented_memory_output_stream<Iterator>;
 public:
     using has_with_stream = std::false_type;
     using iterator_type = Iterator;
     fragmented_memory_input_stream(Iterator it, size_t size)
-        : _it(it), _size(size) {
+        : _it(it), _size(size), _initial_size(size) {
     }
 
+    size_t position() const {
```

Comment:
nit: `const noexcept`
why any reason for not marking this one with `[[gnu::always_inline]]` too?

Reply:
@bhalevy why add this ugly and non-standard `[[gnu::always_inline]]``? What does it add?

Reply:
Without judging if it's helpful or not,
we need to add to add it in both cases or neither.
I think the compiler will inline it anyway when it's defined as a method in the header file so I agree it's not needed,
but that said it's already present in this header.  I wonder why